server/svix-server/src/db/models/application.rs

```diff
@@ -42,21 +49,112 @@ impl Related<super::message::Entity> for Entity {
     }
 }
 
+impl Related<super::applicationmetadata::Entity> for Entity {
+    fn to() -> RelationDef {
+        Relation::Metadata.def()
+    }
+}
+
+impl Model {
+    pub async fn fetch_application_metadata(
+        &self,
+        db: &impl ConnectionTrait,
+    ) -> error::Result<applicationmetadata::ActiveModel> {
+        let query = applicationmetadata::Entity::secure_find(self.id.clone());
+        let metadata = ctx!(query.one(db).await)?
+            .map(applicationmetadata::ActiveModel::from)
+            .unwrap_or_else(|| applicationmetadata::ActiveModel::create(self.id.clone(), None));
+
+        Ok(metadata)
+    }
+
+    pub async fn fetch_many_with_metadata(
+        db: &DatabaseConnection,
+        org_id: OrganizationId,
+        limit: u64,
+        after_id: impl Into<Option<ApplicationId>>,
+    ) -> error::Result<impl Iterator<Item = (Self, Metadata)>> {
+        let mut query = Entity::secure_find(org_id)
+            .order_by_asc(Column::Id)
+            .limit(limit);
+
+        if let Some(id) = after_id.into() {
+            query = query.filter(Column::Id.gt(id))
+        }
+
+        let results = ctx!(
+            query
+                .find_also_related(applicationmetadata::Entity)
+                .all(db)
+                .await
+        )?;
+
+        Ok(results.into_iter().map(|(app, metadata)| {
+            let metadata = metadata.map(|m| m.metadata()).unwrap_or_default();
+            (app, metadata)
+        }))
+    }
+
+    pub async fn fetch_with_metadata(
+        db: &DatabaseConnection,
+        org_id: OrganizationId,
+        id_or_uid: ApplicationIdOrUid,
+    ) -> error::Result<Option<(Self, Metadata)>> {
+        let result = ctx!(Self::fetch_with_application_metadata(db, org_id, id_or_uid).await)?
+            .map(|(app, metadata)| (app, metadata.map(|m| m.metadata()).unwrap_or_default()));
+
+        Ok(result)
+    }
+
+    pub async fn fetch_with_application_metadata(
+        db: &DatabaseConnection,
+        org_id: OrganizationId,
+        id_or_uid: ApplicationIdOrUid,
+    ) -> error::Result<Option<(Self, Option<applicationmetadata::Model>)>> {
+        ctx!(
+            Entity::secure_find_by_id_or_uid(org_id, id_or_uid)
+                .find_also_related(applicationmetadata::Entity)
+                .one(db)
+                .await
+        )
+    }
+}
+
 impl ActiveModelBehavior for ActiveModel {
-    fn new() -> Self {
```

Comment:
It's implemented by default, I opted to remove because we don't use it anywhere and I wanted `fn create` to be the defactor / obvious way to initialize a new app.

Reply:
Yeah, but it's implemented by default without the `created_at` and the things that we want, no? That's why we've used it elsewhere. Maybe let's have it here for good measure, and also use it in `create`? Just to make sure we don't use it by accident.

Reply:
It's noise, and wasted code.

Whether or not we override `::new()`, inserting an `application::ActiveModel::new()` record will _always_ encounter a runtime error because it's missing some required fields (like an org_id).

In other words, overriding ::new() does nothing to protect us against runtime errors. But keeping the function defined just encourages folks to use it over `create(...)`. Why bother with the extra code / complexity?